lib/request_validations/conversions.ex

```diff
@@ -0,0 +1,177 @@
+  @spec parse(String.t(), :boolean) :: {:ok, boolean()} | {:error, :bad_boolean}
+  @spec parse(String.t() | [String.t()], :list) :: {:ok, [String.t()]}
+  @spec parse(String.t(), :string) :: {:ok, String.t()}
+  def parse(value, :date), do: Date.from_iso8601(value)
+  def parse(value, :integer), do: parse_integer(value)
+  def parse(value, :float), do: parse_float(value)
+  def parse(value, :boolean), do: parse_boolean(value)
+  def parse(value, :list), do: parse_list(value)
+  def parse(value, :string), do: parse_string(value)
+  def parse(value, :uuid), do: parse_uuid(value)
+
+  @spec parse_string(term()) :: {:ok, String.t()} | {:error, :bad_string}
+  def parse_string(value) when is_binary(value), do: {:ok, value}
+  def parse_string(_value), do: {:error, :bad_string}
+  @spec parse_integer(term()) :: {:ok, integer()} | {:error, :bad_integer}
+  def parse_integer(value) do
+    case Integer.parse(value) do
+      {value, ""} -> {:ok, value}
+      _error -> {:error, :bad_integer}
+    end
+  end
+
+  @spec parse_float(term()) :: {:ok, float()} | {:error, :bad_float}
+  def parse_float(value) do
+    case Float.parse(value) do
+      {value, ""} -> {:ok, value}
+      _error -> {:error, :bad_float}
+    end
+  end
+
+  @spec parse_boolean(term()) :: {:ok, boolean()} | {:error, :bad_boolean}
+  def parse_boolean("true"), do: {:ok, true}
+  def parse_boolean("false"), do: {:ok, false}
+  def parse_boolean(_unexpeted), do: {:error, :bad_boolean}
+
+  @spec parse_list(String.t() | [String.t()]) :: {:ok, [String.t()]} | {:error, :bad_list}
+  def parse_list(list) when is_list(list), do: {:ok, list}
+  def parse_list(val) when is_binary(val), do: {:ok, String.split(val, ",")}
+  def parse_list(_unexpected), do: {:error, :bad_list}
+
+  def parse_uuid(uuid_string)
+
+  def parse_uuid(
+        <<a1, a2, a3, a4, a5, a6, a7, a8, ?-, b1, b2, b3, b4, ?-, c1, c2, c3, c4, ?-, d1, d2, d3,
+          d4, ?-, e1, e2, e3, e4, e5, e6, e7, e8, e9, e10, e11, e12>>
+      ) do
+    <<d(a1)::4, d(a2)::4, d(a3)::4, d(a4)::4, d(a5)::4, d(a6)::4, d(a7)::4, d(a8)::4, d(b1)::4,
+      d(b2)::4, d(b3)::4, d(b4)::4, d(c1)::4, d(c2)::4, d(c3)::4, d(c4)::4, d(d1)::4, d(d2)::4,
+      d(d3)::4, d(d4)::4, d(e1)::4, d(e2)::4, d(e3)::4, d(e4)::4, d(e5)::4, d(e6)::4, d(e7)::4,
+      d(e8)::4, d(e9)::4, d(e10)::4, d(e11)::4, d(e12)::4>>
+  catch
+    :error -> :error
+  else
+    raw_uuid -> {:ok, raw_uuid}
+  end
+
+  def parse_uuid(_), do: :error
+  defp d(?0), do: 0
+  defp d(?1), do: 1
+  defp d(?2), do: 2
+  defp d(?3), do: 3
+  defp d(?4), do: 4
+  defp d(?5), do: 5
+  defp d(?6), do: 6
+  defp d(?7), do: 7
+  defp d(?8), do: 8
+  defp d(?9), do: 9
+  defp d(?A), do: 10
+  defp d(?B), do: 11
+  defp d(?C), do: 12
+  defp d(?D), do: 13
+  defp d(?E), do: 14
+  defp d(?F), do: 15
+  defp d(?a), do: 10
+  defp d(?b), do: 11
+  defp d(?c), do: 12
+  defp d(?d), do: 13
+  defp d(?e), do: 14
+  defp d(?f), do: 15
+  defp d(_), do: throw(:error)
```

Comment:
Maybe it's better just to use regexp for UUID validations. WDYT?

Reply:
hmm yes that is true